xbmc/cores/VideoPlayer/VideoPlayerTeletext.cpp

```diff
@@ -735,7 +735,7 @@ void CDVDTeletextData::LoadPage(int p, int sp, unsigned char* buffer)
 
 void CDVDTeletextData::ErasePage(int magazine)
 {
-  std::unique_lock<CCriticalSection> lock(m_critSection);
+  std::unique_lock<CCriticalSection> lock(m_TXTCache->m_critSection);
```

Comment:
You don't need to lock in `void CDVDTeletextData::AllocateCache(int magazine)` ?

Reply:
CDVDTeletextData::AllocateCache() is a private function called once in a function already protected by the mutex. The mutex could be added, but this is not required with the current implementation.